pkgs/applications/editors/vscode/generic.nix

```diff
@@ -108,6 +111,18 @@ let
       )
     '';
 
+    # See https://github.com/NixOS/nixpkgs/issues/49643#issuecomment-873853897
+    postPatch = ''
+      # this is a fix for "save as root" functionality
+      packed="resources/app/node_modules.asar"
+      unpacked="resources/app/node_modules"
+      ${nodePackages.asar}/bin/asar extract "$packed" "$unpacked"
```

Comment:
Why not use nativeBuildInputs?

Reply:
I have no idea how to/what is it but you are very welcome to propose a change. I would be happy to learn something new about nix.

Reply:
> I have no idea how to/what is it but you are very welcome to propose a change. I would be happy to learn something new about nix.

If nodePackages.asar is put in nativeBuildInputs then the asar command would be added to the build path so the command would be `asar extract` 

Imo it's fine as it is, no change needed.

Reply:
Not using hardcored paths in build environments is preferred. Hardcore paths are more likely to break and slip by treewide changes related to buildInputs. It is totally fine to use absolute paths in patching/build scripts etc.